srcjs/shinyapp.js

```diff
@@ -404,7 +416,15 @@ var ShinyApp = function() {
         el.data('data-display-if-func', condFunc);
       }
 
-      var show = condFunc(scope);
+      // The data-ns-prefix attribute is always present, but might be empty. If
+      // it's empty, no keys are removed or renamed by narrowScope.
+      var nsPrefix = el.attr('data-ns-prefix');
```

Comment:
Can you check for null/undefined here actually? It's possible for existing HTML templates to have conditional panels hard coded. Our support for such code is not clearly defined but in this case it's so trivial to not break them.